etl/airflow.md

```diff
@@ -0,0 +1,329 @@
+preferred
+solution](https://github.com/datamade/scrapers-us-municipal/blob/c537c27dbd949cf6b838612325f087b2e3860252/scripts/la-metro-crontask)
+at DataMade!
+
+However, cron contributed to or made it more difficult to solve some tough
+problems for our scrapers, including [scheduling
+typos](https://github.com/datamade/scrapers-us-municipal/pull/50),
+[colliding](https://github.com/datamade/scrapers-us-municipal/issues/38)
+[scrapes](https://github.com/datamade/scrapers-us-municipal/issues/3), and
+[indefinitely hung
+processes](https://github.com/datamade/scrapers-us-municipal/issues/55). Beyond
+bugs and developer experience, our usage of cron did not give our clients any
+insight into whether scrapes had run, and to what outcome.
+
+What we wanted was a framework that would allow us to define, manage, and
+monitor our ETL jobs, and also expose an interface for interested parties to
+follow along. We investigated three candidates:
+[Airflow](https://airflow.apache.org/index.html),
+[Luigi](https://github.com/spotify/luigi), and
+[Rundeck](https://www.rundeck.com/open-source). Rundeck seemed optimized more
+for server administration than data workflows, so we mostly considered Airflow
+vs. Luigi.** **
+
+[This article](https://towardsdatascience.com/data-pipelines-luigi-airflow-everything-you-need-to-know-18dc741449b7)
+provides an excellent comparison of Airflow and Luigi. The important
+[distinctions for us were as follows:
+
+1. Airflow supports scheduling tasks, while Luigi [purposefully omits this
+functionality](https://luigi.readthedocs.io/en/stable/central_scheduler.html).
+2. Luigi is a build tool. [Its units of
+work](https://luigi.readthedocs.io/en/stable/tasks.html) are more Make-like in
+that they are tightly coupled to output from their composite targets.
+Conversely, Airflow operators run independently of one another, i.e., Airflow
+feels more conceptually appropriate for scheduling tasks that don’t necessarily
+create file-like artifacts, like scraping data.
+3. Airflow has eclipsed Luigi in
+popularity and maintenance resources, as measured by forks, stars, and watchers,
+and commits, respectively.
+
+Based on these impressions, and [an initial proof of
+concept](https://github.com/datamade/scrapers-airflow), we selected Airflow as
+our tool of choice for ETL management. Our first Airflow instance is the LA
+Metro Dashboard, which has been open sourced
+[here](https://github.com/datamade/la-metro-dashboard).
+
+## Basic concepts
+
+Airflow is a framework that relies on several key concepts. For a tutorial that
+guides you through a summary of these concepts in Airflow’s words, go to
+[https://airflow.apache.org/docs/stable/tutorial.html#example-pipeline-definition](https://airflow.apache.org/docs/stable/tutorial.html#example-pipeline-definition).
+Below are listed the most important Airflow concepts based on our work with
+Airflow thus far.
+
+### DAG (directed acyclic graph)
```

Comment:
@fatima3558 What do you think about reorganizing these concept subheds to follow the format:

### Concept

**Airflow's description**: Link

[Discussion]

#### Notes

- ...